app/src/main/java/de/simon/dankelmann/bluetoothlespam/Dialogs/PersistentLocationPermissionDialog.kt

```diff
@@ -0,0 +1,307 @@
+        
+        grantButton.text = "Allow All The Time"
+        grantButton.setOnClickListener {
+            // Request background permission
+            backgroundPermissionLauncher.launch(Manifest.permission.ACCESS_BACKGROUND_LOCATION)
+        }
+        
+        settingsButton.text = "Close App"
+        settingsButton.setOnClickListener {
+            activity.finishAffinity()
+        }
+        
+        // Show dialog and make it persistent
+        dialog!!.show()
+        
+        // Make sure dialog stays on top
+        dialog!!.window?.apply {
+            setFlags(
+                WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON,
+                WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
+            )
+        }
+    }
+    
+    private fun openAppSettings() {
+        Log.d(TAG, "Opening app settings button clicked")
+        // For Android 10 (Q) and above, try to open location settings directly
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            try {
+                // Try to use a more direct intent for location permissions
+                val intent = Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS)
+                val uri = Uri.fromParts("package", activity.packageName, null)
+                intent.data = uri
+                
+                // Add flags to ensure it opens in a new task and clears previous tasks
+                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                
+                // Update dialog message to guide the user
+                val messageTextView = dialog!!.findViewById<TextView>(R.id.tv_permission_message)
+                messageTextView?.text = "Please tap 'Permissions' → 'Location' → select 'Allow all the time' option in the settings screen that will open."
+                
+                // Start the activity and log the action
+                Log.d(TAG, "Launching application details settings")
+                activity.startActivity(intent)
+                
+                // Don't dismiss the dialog yet - we'll check permission on resume
+            } catch (e: Exception) {
+                // Fallback to regular app settings if specific intent fails
+                Log.e(TAG, "Failed to open specific location settings: ${e.message}")
+                openRegularAppSettings()
+            }
+        } else {
+            // For older versions, use the regular app settings
+            openRegularAppSettings()
+        }
+    }
+    
+    private fun openRegularAppSettings() {
+        // Open app-specific permission settings (fallback method)
+        val intent = Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS)
+        val uri = Uri.fromParts("package", activity.packageName, null)
+        intent.data = uri
+        activity.startActivity(intent)
+    }
+    
+    private fun dismissDialog() {
+        dialog?.dismiss()
+        dialog = null
+    }
+    
+    fun onResume() {
```

Comment:
This function just duplicates some if the logic that is already in checkAndRequestLocationPermission(). There should be only a single place for this logic.